File: Java/0101-0150/125. Valid Palindrome.java
// https://leetcode.com/problems/valid-palindrome/

class Solution {
    public boolean isPalindrome(String s) {
        s = s.toLowerCase();
        //remove non-alphabet characters
        StringBuilder palindrome = new StringBuilder();

        for(char letter : s.toCharArray()){
            if((letter >= 'a' && letter <= 'z') || (letter >= '0' && letter <= '9'))
                palindrome.append(letter);
        }

        // Two-Pointers
        int first = 0, second = palindrome.length()-1;

        while(first <= second){
            if(palindrome.charAt(first++) != palindrome.charAt(second--))
                return false;
        }

        return true; // Its a palindrome
    }
}

File: Python/0101-0150/125. Valid Palindrome.py
# https://leetcode.com/problems/valid-palindrome/

class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = s.lower()
        # remove non-alphabet characters
        palindrome = ""

        for letter in s:
            if((letter >= 'a' and letter <= 'z') or (letter >= '0' and letter <= '9')):
                palindrome += letter

        # Two-Pointers
        first, second = 0, len(palindrome)-1

        while first <= second:
            if palindrome[first] != palindrome[second]:
                return False
            first += 1
            second -= 1

        return True # Its a palindrome
